rewrite-java/src/main/java/org/openrewrite/java/tree/J.java

```diff
@@ -332,9 +332,22 @@ final class ArrayType implements J, TypeTree, Expression {
         @With
         TypeTree elementType;
 
+        @Deprecated
+        @Nullable
```

Comment:
I added a comment on the commit rather than the PR. I suggest we remove all deprecated fields and instead add a JsonCreator constructor with some mapping logic in it.